File: csharp/unittests/policy.tests/x509/SubjectPublicKeySizeField_InjectReferenceValueTest.cs
using System;
using System.Security.Cryptography.X509Certificates;
using FluentAssertions;
using Health.Direct.Policy.X509;
using Xunit;

namespace Health.Direct.Policy.Tests.x509
{
    public class SubjectPublicKeySizeField_InjectReferenceValueTest
    {
        [Fact]
        public void SubjectPublicKeySizeField_injectReferenceValueTest()
        {
            var cert = new X509Certificate2(@"resources/certs/altNameOnly.der");
            var field = new SubjectPublicKeySizeField();
            field.InjectReferenceValue(cert);
            field.GetPolicyValue().GetPolicyValue().Should().Be(1024);
        }

        [Fact]
        public void testInjectRefereneValue_rsa2024_assertValue()
        {
            var cert = new X509Certificate2(@"resources/certs/umesh.der");
            var field = new SubjectPublicKeySizeField();
            field.InjectReferenceValue(cert);
            field.GetPolicyValue().GetPolicyValue().Should().Be(2024);
        }

        [Fact]
        public void testInjectRefereneValue_dsa1024_assertValue()
        {
            var cert = new X509Certificate2(@"resources/certs/dsa1024.der");
            var field = new SubjectPublicKeySizeField();
            field.InjectReferenceValue(cert);
            field.GetPolicyValue().GetPolicyValue().Should().Be(1024);
        }

        [Fact]
        public void testInjectRefereneValue_ecc_assertValue()
        {
            var cert = new X509Certificate2(@"resources/certs/ecc.der");
            var field = new SubjectPublicKeySizeField();
            field.InjectReferenceValue(cert);
            field.GetPolicyValue().GetPolicyValue().Should().Be(0);
        }

        [Fact]
        public void TestInjectReferenceValue_NoInjection_GetPolicyValue_AssertException()
        {
            var field = new SubjectPublicKeyAlgorithmField();
            Action action = () => field.GetPolicyValue();
            action.ShouldThrow<InvalidOperationException>();
        }
    }
}

File: java/direct-policy/src/test/java/org/nhindirect/policy/x509/SubjectPublicKeySizeField_injectReferenceValueTest.java
package org.nhindirect.policy.x509;

import java.security.cert.X509Certificate;

import org.nhindirect.policy.util.TestUtils;
import org.nhindirect.policy.x509.SubjectPublicKeyAlgorithmField;
import org.nhindirect.policy.x509.SubjectPublicKeySizeField;

import junit.framework.TestCase;

public class SubjectPublicKeySizeField_injectReferenceValueTest extends TestCase
{
	public void testInjectRefereneValue_rsa1024_assertValue() throws Exception
	{
		final X509Certificate cert = TestUtils.loadCertificate("altNameOnly.der");
		
		final SubjectPublicKeySizeField field = new SubjectPublicKeySizeField();
		
		field.injectReferenceValue(cert);
		
		final int value = field.getPolicyValue().getPolicyValue();
		
		assertEquals(1024, value);
		
	}	
	
	public void testInjectRefereneValue_rsa2024_assertValue() throws Exception
	{
		final X509Certificate cert = TestUtils.loadCertificate("umesh.der");
		
		final SubjectPublicKeySizeField field = new SubjectPublicKeySizeField();
		
		field.injectReferenceValue(cert);
		
		final int value = field.getPolicyValue().getPolicyValue();
		
		assertEquals(2024, value);
		
	}	
	
	public void testInjectRefereneValue_dsa1024_assertValue() throws Exception
	{
		final X509Certificate cert = TestUtils.loadCertificate("dsa1024.der");
		
		final SubjectPublicKeySizeField field = new SubjectPublicKeySizeField();
		
		field.injectReferenceValue(cert);
		
		final int value = field.getPolicyValue().getPolicyValue();
		
		assertEquals(1024, value);
		
	}	
	
	public void testInjectRefereneValue_ecc_assertValue() throws Exception
	{
		final X509Certificate cert = TestUtils.loadCertificate("ecc.der");
		
		final SubjectPublicKeySizeField field = new SubjectPublicKeySizeField();
		
		field.injectReferenceValue(cert);
		
		final int value = field.getPolicyValue().getPolicyValue();
		
		assertEquals(0, value);
		
	}	
	
	public void testInjectRefereneValue_noInjection_getPolicyValue_assertException() throws Exception
	{
		
		final SubjectPublicKeyAlgorithmField field = new SubjectPublicKeyAlgorithmField();
		
		boolean exceptionOccured = false;
		
		try
		{
			field.getPolicyValue();
		}
		catch (IllegalStateException e)
		{
			exceptionOccured = true;
		}
		assertTrue(exceptionOccured);
	}
}
